crates/ruff/src/rules/flake8_async/rules.rs

```diff
@@ -0,0 +1,224 @@
+/// ```
+#[violation]
+pub struct OpenSleepOrSubprocessInAsyncFunction;
+
+impl Violation for OpenSleepOrSubprocessInAsyncFunction {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Async functions should not call `open`, `time.sleep`, or `subprocess` methods")
+    }
+}
+
+const OPEN_SLEEP_OR_SUBPROCESS_CALL: &[&[&str]] = &[
+    &["", "open"],
+    &["time", "sleep"],
+    &["subprocess", "run"],
+    &["subprocess", "Popen"],
+    // Deprecated subprocess calls:
+    &["subprocess", "call"],
+    &["subprocess", "check_call"],
+    &["subprocess", "check_output"],
+    &["subprocess", "getoutput"],
+    &["subprocess", "getstatusoutput"],
+    &["os", "wait"],
+    &["os", "wait3"],
+    &["os", "wait4"],
+    &["os", "waitid"],
+    &["os", "waitpid"],
+];
+
+/// ASY101
+pub(crate) fn open_sleep_or_subprocess_call(checker: &mut Checker, expr: &Expr) {
+    if in_async_function(&checker.ctx) {
+        if let ExprKind::Call(ast::ExprCall { func, .. }) = &expr.node {
+            if let Some(call_path) = checker.ctx.resolve_call_path(func) {
+                if OPEN_SLEEP_OR_SUBPROCESS_CALL.contains(&call_path.as_slice()) {
+                    checker.diagnostics.push(Diagnostic::new(
+                        OpenSleepOrSubprocessInAsyncFunction,
+                        func.range,
+                    ));
+                }
+            }
+        }
+    }
+}
+
+/// ## What it does
+/// Checks that async functions do not contain calls to blocking synchronous
+/// process calls via the `os` module.
+///
+/// ## Why is this bad?
+/// Blocking an async function via a blocking call will block the entire
+/// event loop, preventing it from executing other tasks while waiting for the
+/// call to complete, negating the benefits of asynchronous programming.
+///
+/// Instead of making a blocking call, use an equivalent asynchronous library
+/// or function.
+///
+/// ## Example
+/// ```python
+/// async def foo():
+///     os.popen()
+/// ```
+///
+/// Use instead:
+/// ```python
+/// def foo():
+///     os.popen()
+/// ```
+#[violation]
+pub struct BlockingOsCallInAsyncFunction;
+
+impl Violation for BlockingOsCallInAsyncFunction {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Async functions should not call synchronous `os` methods")
+    }
+}
+
+const UNSAFE_OS_METHODS: &[&[&str]] = &[
+    &["os", "popen"],
+    &["os", "posix_spawn"],
+    &["os", "posix_spawnp"],
+    &["os", "spawnl"],
+    &["os", "spawnle"],
+    &["os", "spawnlp"],
+    &["os", "spawnlpe"],
+    &["os", "spawnv"],
+    &["os", "spawnve"],
+    &["os", "spawnvp"],
+    &["os", "spawnvpe"],
+    &["os", "system"],
+];
+
+/// ASY102
+pub(crate) fn blocking_os_call(checker: &mut Checker, expr: &Expr) {
+    if in_async_function(&checker.ctx) {
+        if let ExprKind::Call(ast::ExprCall { func, .. }) = &expr.node {
+            if let Some(call_path) = checker.ctx.resolve_call_path(func) {
```

Comment:
@qdegraaf - I changed this to use `resolve_call_path`, which will automatically detect situations like `from os import system` or even `from os import system as foo` or `import os as foo; foo.system`.